prometheus/metrics.go

```diff
@@ -109,22 +109,31 @@ func getServiceMetricsAsync(api v1.API, namespace string, servicename string, du
 		return chin, chout
 	}
 
-	rqcountinCh, rqcountoutCh := fetchRateAsync("istio_request_count")
-	rqsizeinCh, rqsizeoutCh := fetchHistoAsync("istio_request_size")
-	rqdurinCh, rqduroutCh := fetchHistoAsync("istio_request_duration")
-	rssizeinCh, rssizeoutCh := fetchHistoAsync("istio_response_size")
-
 	metrics := make(map[string]Metric)
 	histograms := make(map[string]Histogram)
 
+	rqcountinCh, rqcountoutCh := fetchRateAsync("istio_request_count", labelsIn, labelsOut)
+	if component == "details" {
+		rqsizeinCh, rqsizeoutCh := fetchHistoAsync("istio_request_size")
+		rqdurinCh, rqduroutCh := fetchHistoAsync("istio_request_duration")
+		rssizeinCh, rssizeoutCh := fetchHistoAsync("istio_response_size")
+
+		histograms["request_size_in"] = <-rqsizeinCh
+		histograms["request_size_out"] = <-rqsizeoutCh
+		histograms["request_duration_in"] = <-rqdurinCh
+		histograms["request_duration_out"] = <-rqduroutCh
+		histograms["response_size_in"] = <-rssizeinCh
+		histograms["response_size_out"] = <-rssizeoutCh
+	} else {
+		labelsErrorsIn := fmt.Sprintf("{destination_service=\"%s.%s.%s\",response_code=~\"5.*\"}", servicename, namespace, clustername)
+		labelsErrorsOut := fmt.Sprintf("{source_service=\"%s.%s.%s\",response_code=~\"5.*\"}", servicename, namespace, clustername)
```

Comment:
Do we not want more than just 5XX errors? What about 4XX?

Reply:
Ok, I'll check how I can include them